### ISSUE TITLE
Adding co-authors, citation format change and justify text not working

### ISSUE THREAD

**Issue by Annotuser** [Annotum/Annotum#248]
Hi,

I am trying to use Annotum but facing some issues for the citation format, adding co-authors and justifying text in the Article editor (but not in the post type).
I am not highly expert in coding but here is a brief description of my issues: 

1) Citation format: In the function template.php file, I am trying to slightly modify the citation format by adding the name of author that makes the modification, so I added “modified by & $author” in the following code line but there is no way the change I’m making takes effect. Here is how I proceed:

In template.php: 

                                                _x('%1$s. %2$s. %4$s. %5$s [last modified: %6$s by %1$s]. Edition %3$s. %7$s', 'Citation format', 'anno'),

                                                $authors, // 1

                                                $title, // 2

                                                $edition, // 3

                                                $site, // 4

                                                $post_date, // 5

                                                $last_modified, // 6

I am simply adding ‘by %1$s’ or ‘by $authors’ in the code piece [last modified:  %6$s by $authors] but nothing happens. Actually, whatever the changes I make here, the citation format remains unchanged or causes errors.

2) I would like to place the citation format in the upper or the bottom part of the article (just below the abstract) and when I tried to place the code above in article-post-type.php, it does not work but only error messages! Did I place the right code in the right place? 

3) Co-author issue: The author’s name does not appear in the citation format at all, even if it was the admin!
Besides, when I added other members as writers, contributors or subscribers…etc., the name of the admin always remains as the author and no other way to change it to another author or co-authors! Apparently, it is not possible to display co-authors to any kind of contributions. 

4) It is possible to justify texts using the ‘TinyMCE Advanced’ plugin only if texts are added as Posts but not as Articles! The justification of text seems not working in the Article editor but only for Post types. 

Any tips to solve these issues would be greatly appreciated. 

Thank you,
Best,
Annotum user

**Comment by Annotuser** [comment]
Well, for the co-author issues, I got it fixed by activating the workflow option and adding authors as members 'writers' and then select from a dropdown menu. Not so handy but it is still OK, except for the authors' information that is not displayed. We need to hover over to see them. 
The most critical issue anyway is the citation format and the justification of text in the Article editor. 
I am still unable to get the text justified and to change to citation format.
I am using the latest Wordpress version (4.6) and latest Annotum (2.1.1) on localhost installation and I also tried to use older versions of both of them but the issues remain unsolved.  
**For the citation format:** I am trying to simplify it for example to the following format: 
"$author, $title, $site, $date: F Y D" 
But I am still facing many issues and errors whatever the change or tweak I make in the files template.php and tool-bar.phb. I am certainy using a wrong approach or wrong code. 
So, I really would appreciate if somebody can advise how to get this issue resolved to allow me continue testing Annotum and explore its other functionalities. 
I just need to eventually make a citation format as simple as the following: 
**Author, article title, site name, date (year, month, day, Hours**
What would be the code and where should I place it?
And for the option of text justifying, why it does not work in the article editor of TinyMCE? 
Actually, it does justify the text only during the writing but I click to save or publish the text it turns back into an unjustified paragraph!

Thank you again for any advice. 

Regards,
Annotuser.

**Comment by carlthewebmaster** [comment]
For better or worse, Annotum very vigorously enforces strict adherence to
the NLM-JATS XML schema, meaning any layout markup beyond what the toolbar
provides is forcibly removed by the editor. That means that you cannot save
fully-justified text as an attribute of the article content.

However you should be able to modify the display on site by changing the
css.

**Comment by Annotuser** [comment]
Thank you for your input.
Having the text justified would help optimizing the reading process in addition to be an esthetic feature to save space and margins. 
I can read justified texts more flawlessly than unjustified ones. 
If you take a look at unjustified texts (the comments here for example), you will see how mush space is lost or unused at the right. 
I do not see why justified texts are not appropriate for NLM-JATS XML schema? 
By the way, the one-line format of the main css file in Annotum makes it a little bit hard to read and find the appropriate values to customize. 
This issue, however, remains tiny compared with the citation format that should be more easily customizable and why not visible in the text rather than to be hidden in a citation cell or a call out link. 
I am still unable to find a way to tweak it as I wish, for example as simple and informative as follows:

**Author, title, site, date 'year, month, day, hour'** 

That's all.
This is in my view the easiest and most efficient way to cite. It is easy to remember and easy to read and to understand. 
Another interesting option would be to offer the possibility to download the citation as a file readable in a bibliography manager (Endnote, Zotero, etc.) but not only to see it. 
Having the citation format easily comprehensible and visible in the page would save the reader's time not to making additional moves or clicks. 

Thank you once again for your feedback and for any tip to get this issue solved/improved. I'll appreciate it.

**Comment by carlthewebmaster** [comment]
XML-JATS only supports structural markup, not layout (e.g. justification) which is left to the publisher.  In the case of Annotum, the publisher is the site owner, so you can implement this display format by:

- Using Jetpack's custom CSS feature
- Modifying the base css for the Annotum theme (this can be tricky since the file is minimized as you have seen)
- Setting up a child theme with the formatting you want

In either case, adding ```text-align: justify;``` to the CSS for the ```<p>``` element will get you the result you want.

**Comment by Annotuser** [comment]
Thank you, Carl, for your valuable comments, highly appreciated.
A child theme may indeed be an option to consider, though having such a function (text justification) built in the article editor directly would be easier and more efficient to stick with the WYSIWYG rule. 
I did not try the suggestions yet, but I am afraid that it may cause some unexpected problems when for example complex texts and graphics (text + images+ table + equations... etc.) are added. This is why a built-in function for text justification/un-justification may be more efficient to visually control the final output of the layout and to see how it looks like (in preview) before a final publishing.  
Anyway, and if I may, what about the customization of the citation format? 
This is the third time I refer to but, if I may again, you seem overlooking it? ;) 
Why it is so hard to modify it? What an open source would mean then? Does it mean that one should be developer or highly skilled in IT to make simple changes or customizations?  
I have seen some sites that have already modified their citation style by adding for example DOI serial numbers at the end of the default format:
https://journal.emergentpublications.com
So, shouldn’t be easy to modify or customize this format as I previously mentioned (i.e., author, title, site, date)? 
Why any change in the order or the elements of the default citation format in ‘template.php’ does not work?
However, by comparing the default citation elements, their order in 'template.php' with the citation output, it seems that the template.php should be the right place to workaround to customize the citation format but this does not seem working, which is somewhat disappointing. 
Endnote contains thousands of citation styles and it allows to modify/customize or add any new desired output styles. Why Annotum, as an open source, does not allow it by imposing only one 'recalcitrant' citation format? 
Shouldn’t Annotum (as an open source) be more customizable and flexible than Endnote or any other "closed code source soft"?!  

Regards,

**Comment by carlthewebmaster** [comment]
Final note regarding text justification.  Just as with other sorts of display formatting (e.g., fonts, colors, specific text sizes), text justification is not an inherent part of the article itself.  Annotum does not store or save such formatting information - as with a "real" manuscript submitted to a publisher, the formatting decisions are left to the typesetter, which in the case of Annotum corresponds to the CSS for front-end display.

Regarding variable citation styles, it would indeed be a nice feature to support, but unfortunately it is not within the current scope for Annotum to support this as a feature.

> Anyway, and if I may, what about the customization of the citation format? 
> This is the third time I refer to but, if I may again, you seem 'overlooking' it? ;) 
> Why it is so hard to modify it? What an open source would mean then? Does it mean that one should be developer or highly skilled in IT to make simple changes or customizations?

I absolutely do not mean to overlook your question - it just requires a bit more thought to provide you with a complete response! 

I would answer by saying that in order for it to be a worthwhile addition to Annotum, support for custom citations would have to be as flexible and simple as you suggest.  However writing the support for something simple for the user, that is, adding a flexible, easy-to-use citation formatter that supports even a limited set of custom citation formats, is far from a simple task.  Programming takes time and costs money - in fact most of Annotum's development was funded by interested third parties; it was not a purely volunteer effort (although current support for Annotum is entirely _pro bono_ :) ).

I would suggest that Annotum is in fact far more customizable than closed source software for the simple reason that the source code is available right here on Github for anyone to see, download, or modify.  However that doesn't mean that the modifications are necessarily easy to do - it merely means that one can make them if desired. 

You might want to contact Emergent Publications to ask about their modifications - or to encourage them to submit a pull request to this repository so that their changes can be included in the main Annotum product.

**Comment by Annotuser** [comment]
For the text justification, I think it is not a question of where or who should do it and at which stage of the publishing process, but the existing justification option in WP is not working in Annotum. As we already have a text justification option in WP (via TinyMCE for instance), we do not need to reinvent the wheel, but this option is not working at all in Annotum, though it works well in other themes. So, the problem comes undoubtedly from Annotum. 
For the citation format, and if I may, it should be simple for the theme developer than for non-specialists or non-developers. Of course, it would take time to make a theme and develop it, but the issue I am asking for should be simple as it is already available and easily modifiable, isn't it? Moreover, the theme is open source (no payment option to eventually help the developer). If we could make donations or if the theme was paid, maybe users would pay for it, but in this case, the theme should provide much more 'pro functions' that answer more requirement for scholarly publishing. Personally, I could make some donations if the theme was answering my expectations in managing and publishing articles in a good way (concerning for example the issues I am talking about here; text justification, more flexible citation format, maybe submission by file uploading, etc.). I liked the concept and cleanness of Annotum, but the themes does not seem to be as open as it is presented.   
As the code is Open Source, for sure it is modifiable but for this, one should learn or have relevant skills in PHP coding, which is not the case for all people.  
For me, when we make something, we should make it as good as possible, no matter if it is free or not. At least, it could be made payable but with helpful services for users to answer their needs, otherwise I do not see how user could help developers for their time and efforts spent to developing their tools. Developers may need to bear in mind that users could make donations for free tools they use provided these tools satisfy their needs at best. This is the only way that all parties (developers of free tools and their users).      
I sent a request to the journal in question but I'm not sure they will respond.

**Comment by Annotuser** [comment]
Well, to be candid, after some days of trying, installing and buying some plugins, I come to the conclusion that the code of Annotum is obscure and recalcitrant for real customization, which makes it inappropriate for a good journal. 
I bought some Pro Plugins to eventually compensate many lacking functions in Annotum, but they are not working at all in Annotum. By contrast, they work just fine in other themes.
Even a PHP developer guy wasn't able to make simple tweaks. 
I do not see what is the purpose of the theme's developer to develop an obscurely coded theme if s/he wants people use it? How people would use a tool if it does not answer their needs properly?   
Worse, Annotum conceals many basic functions of WordPress without necessarily adding new ones. 
Annotum would suit blogging or narration tales for students or juniors, but not for a professional and serious journal, particularly in the absence of third party plugins to add the basic function of a modern journal. 
Using other easily customizable themes with some pro plugins should make much better use of WP functionalities than with a recalcitrant and not upgradable or updatable theme (Annotum).

**Comment by carlthewebmaster** [comment]
I'm sorry that Annotum doesn't meet your needs.  There are some "professional and serious" journals that have found it useful and continue to use it, but of course that doesn't mean it will work for everyone.

In your earlier post, you asked:

>  Isn't [the goal] to make it as good and customizable [as] possible?

Well, my goal with Annotum was certainly to make it as good as possible, _within the time and budget limitations in place_.  I don't necessarily agree that making it "as customizable as possible" is always a good goal, because, and we can take custom citation formats as an example here, maximum customization adds significantly to the coding and testing cost, and can make the Annotum more complicated and difficult to use.  While it can be limiting, sometimes simpler is better.

In any case, I wish you success in your Journal publishing efforts, and I thank your for your detailed feedback and for giving Annotum a chance.

**Comment by Annotuser** [comment]
Thank you for your comment, and sorry for my frankness and critical comment above. This is because the plugins I tested and paid for do not work with Annotum.  
I still do not see the link between obscuring a theme code and the time and budget required for it? 
In contrary, I think that making a theme simple and easily customizable require much less efforts and time than making it hardly customizable with a complex and obscure code. 
If budget is a 'major' issue not to make Annotum as "good" as possible, why to make it free then?
You could make it either paid or you develop two versions as do many themes' and plugins' developers: one free version with basic functions and another for more customizable and rich functions disabled in the basic/free version. Users can then test and have an overview of the potential functions of the theme and decide to buy it or not. Whatever the money you get from selling a pro version, you will be 'winner' than with a free version that wouldn't be used by so many people. 
Otherwise, what do you benefit from making a theme free and hard to customize?!!! 
Let's imagine the two following possibilities:
1) A feature-rich theme and easy customizable to include missing options, 
2) A theme with a few basic options and hardly customizable (as is the case of Annotum now).
Which theme people would use the most?
Isn't the first one? 

Do you have some 'serious' examples of growing journals using Annotum? 
From my side, except the above example I contacted for DOI issue (but haven't got any response), I did not find any 'serious' or well-established journal based on Annotum. By contrast, there are many based on WP with other themes.  
I think that the reasons behind the low rate of using Annotum could be at least in part the issues I am talking here and the difficulty to customize it. 
Annotum lacks important functions and options that users cannot add by either third party plugins or by self-tweaks, which makes it unpopular for someone who wishes to make a real journal.
Science mag has recently switched to use Drupal for its online publishing platform. Why they did not use WordPress and Annotum?
I think because it is not as customizable as they wish (or at least, not as easily customizable as they wish). 
I think you are missing a great potential to popularize your theme by obscuring its code and limiting it to basics options only and by rendering it incompatible with other plugins that could eventually fulfill the options lacking in Annotum.  
In other words, Annotum lacks important options and functions for scholarly journal, and at the same time, it is incompatible with other plugins that might fulfill them. 
This is a serious drawback that turns many people off using it. 
I hope that my critical comments would be helpful for you to improve the theme.

**Comment by Annotuser** [comment]
Before switching to another theme because Annotum falls short to answer basic requirements of a journal manager, despite the goodness and cleanness of the theme, here are some last remarks to eventually help improve the theme:
- Impossible to add or customize a footer,
- Impossible to customize a webpage to contain more than a column (only the default design is possible),
- Impossible to add articles as downloadable PDF files (however, this is a basic function available in most WP themes that allows to add any media type including clickable or downloadable PDF file, but this function is surprisingly absent in Annotum!). 
- Absence of updates and support.

**Comment by mac2net** [comment]
I would suggest to @Annotuser to come down on one side or another simple or feature full.
Personally I prefer simple - it is easier fo the developer to support and for the site owner supplement with best of breed plugins.

IMO right now Annotum is way too complex. I would suggest to start again with a new theme and take out the weird editor stuff to start.

**Comment by carlthewebmaster** [comment]
> I would suggest to @Annotuser to come down on one side or another simple or feature full.

Great point @mac2net.  The trick with Annotum was that both the simple editorial/peer review workflow and strict XML conformance were 100% must-have features for the people who funded development.  XML conformance in particular is not only challenging from a technical perspective (e.g. lots of custom TinyMCE plugin code that is very sensitive to TinyMCE updates or changes the underlying WordPress code for editing posts), but also very restrictive in terms of what is allowed on the display side.  As a structural markup specification, JATS-XML does not provide style or formatting by its very design, meaning that ordinary WordPress functions such as justified text are not available - they simply can't be stored in the underlying data format for articles.

At the same time, more elaborate academic features such as multicolumn layout, true citation management, cross-references, etc. are beyond the abilities of TinyMCE and would require LaTeX or something like [Authorea](https://www.authorea.com).

It may be that Annotum is reaching the practical end of development and to be honest I am exploring possible exit strategies (e.g. converting Annotum articles to other formats suitable for use in WordPress, Authorea, [OJS](https://pkp.sfu.ca/ojs/), or other systems).

**Comment by mac2net** [comment]
Let me be straight forward - I made a site called 5DOC.org which is proposing to replace PDF and EPUB with offline HTML. Basically make it once and use it on and offline.
I have many samples including a couple of academic articles.
In my analysis, open access folks hate PDF but need to offer offline content. 5DOC gives them an alternative now. This is a very complex subject involving politics of the IDPF and W3C.
My site use WP to manage CSS and JS per post to achieve the flexibility to make different kinds of samples.
Your theme could be modernised to an HTML base, keeping the workflow stuff and meta data, but making customisation much easier. But I do think there is a potentially big market for WordPress in academic publishing.

**Comment by Annotuser** [comment]
Having tried OJS, I honestly think that Annotum could perform better with only minor updates/upgrades to compensate the missing options and to add a few functions required for a journal to appear "professional". Although the new version of OJS (OJS3) comes with serious improvements, Annotum has a great potential if it is just a little more flexible and customizable for the several minor issues raised here. 
For me, if Annotum was offering the following options/functions, it would be much superior to any other manuscript available so far:
1.	A more customizable citation format for a professional citation look (for e.g., author name, title, journal, date) but not with ‘version number’ and ‘brackets’. The possibility to improve the citation format should be simple as the current format already contains the required elements of a professional citation, so it just needs some tweaks to reorder the elements already existing and remove the “brackets” and “version number” from the format. If Annotum is used as a journal, the reader does not care much about the version number, does he?     
2.	Text justification; this should also be a matter of simple tweaking because it is already available in TinyMCE (alternatively, it could be built-in Annotum in the Article editor with a simple code line text justification: align). 
3.	Possibility to upload/download PDF files (PDF files are still popular to be saved and read offline).  
Only these three ‘simple’ functions/options would make Annotum as a serious competitor for other platforms. The options 2 and 3 are already available in WP but unfortunately not working in WP! So, I think that fixing them should be “easy” just to make them compatible or functioning in Annotum. Will such options have something to do with XML? 
I agree with @mac2net, the theme could be modernized on the basis of HTML. If XML is a real requirement, why it should prevent from adding other features such as the possibility to change the citation format or text justification or attaching a PDF file to an article? Does really XML coding prevent from adding or enabling such options? 
I am not a developer but I do not see why and how an XML files (or any other coding) would prevent the optimization of a theme by adding simple but important functions for a journal.

**Comment by Annotuser** [comment]
I have to say that text-justification issue would be much less problematic if it was possible to attach a PDF file (containing the text justified, with the desired page layout and setup). So, practically it remains the citation format and the possibility to attach PDF files that could boost Annotum a little forward.
@carlthewebmaster; don't give up! A little of improvement could make Annotum a much better tool.

**Comment by carlthewebmaster** [comment]
Regarding PDFs, you can attach a pdf as media (a figure) by uploading the PDF using the media editor, then adding a pdf icon as a figure.  Set the link in the figure options to point to the uploaded pdf.

Regarding various update suggestions:

>  Text justification; this should also be a matter of simple tweaking 

> Annotum could perform better with only minor updates/upgrades

These changes are not minor, and I have no current plans to implement them. If you would like them added to Annotum, feel free to fork this repository and implement away.

> Does really XML coding prevent from adding or enabling such options? 

It does not prevent it, but it makes it a lot harder. And again there are no plans for any such changes.

**Comment by carlthewebmaster** [comment]
Clearer example for attaching a PDF version.
1. Open article
2. Insert figure
3. Click "Upload files" and upload the pdf file (here, 'annotum.pdf')
4. Still in the dialog, click upload files and upload a pdf icon.
5. Fill in the attachment details (title, Fig. caption, display type = figure, link to Custom URL)
6. Set the custom url as the absolute link to the PDF file you uploaded in step 3.
7. Click ok and save.
![image](https://cloud.githubusercontent.com/assets/294757/19168409/069756a0-8bde-11e6-8f1e-58d5e099c019.png)

Here's how it looks on the front end after saving:
![image](https://cloud.githubusercontent.com/assets/294757/19168419/16f531ca-8bde-11e6-8787-2e76ae49c526.png)

**Comment by Annotuser** [comment]
Super! This sounds fine! Thank you a lot! 
With this tweak, it is possible indeed to add a PDF link but only in the article's body (or at the end of the text). If, however, the PDF link could be added always in the same place in the page (for e.g., in the subtitle field or immediately after the abstract or any other fixed position in the page for all the articles, it would be nice.  
Wouldn't you have a similar good tweak for the citation format to rearrange it for example by "Author, title, site name, date" (without the current tags of "version number" "last modified", "edition" and the brackets []?
As the citation elements (author's name, title, journal and date) are already available in the default citation format, I would say that rearranging them and removing the unneeded elements (i.e., "edition", "brackets" and "last modified" tags) could be relatively easy for someone who added them and knows well where they are with their dependency.  
In other words, I just would like to use the same citation elements already available but reordered differently from the default option while removing the little unnecessary information (edition's date and number, and brackets).

**Comment by Annotuser** [comment]
Dear carlthewebmaster,
This is just a gentle reminder about my question above on the citation format as I tried to play around but couldn't figure it correctly. So, can you please tell me how re-organize the default citation format to what I stated above (i.e. author, title, site name, date) without the brackets and editions' number?
I appreciate your suggestions. 
Thank you

**Comment by Annotuser** [comment]
Thank you for your comment.
It worked but irreversibly apparently! When I made some changes, they take effect but I couldn't come back to the default or previous setting even after I wait for one hour, 24h or 48h! Is this normal? 
If I may, why should waiting for one hour? Shouldn't we able to remove this limit or at least reduce to a few minutes only, say 10 or 5 min?

Thank you,
